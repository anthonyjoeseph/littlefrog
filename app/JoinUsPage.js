// app/JoinUsPage.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View
} from 'react-native';

import { Actions } from 'react-native-router-flux';

class JoinUsPage extends Component {
  render() {
    return (
      <View style={styles.background}>
        <Text onPress={() => Actions.pop()}>
          Nice it works!
        </Text>
      </View>
    );
  }
}


var styles = StyleSheet.create({
  background: {
    position: 'relative',
    backgroundColor: 'green'
  }
});

export default JoinUsPage;
