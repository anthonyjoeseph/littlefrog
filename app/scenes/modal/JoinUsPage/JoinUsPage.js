// app/JoinUsPage.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  TextInput
} from 'react-native';

import Button from 'react-native-button';

import DismissableFullscreenPopup from '../DismissableFullscreenPopup'

import { Actions } from 'react-native-router-flux';

class JoinUsPage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      emailTextInput: 'example@expl.com'
    };
  }

  render() {
    return (
      <DismissableFullscreenPopup
        onDismiss={() => {Actions.pop();}}>
        <View style={styles.popupView}>
          <View style={styles.topView}>
            <Text style={styles.topText}>Join for Promotions!</Text>
          </View>
          <View style={styles.bottomView}>
            <Text style={styles.bottomText}>Enter your email</Text>
            <TextInput
              style={{height: 40}}
              editable = {true}/>
              <Button
                containerStyle={{padding:5, height:45, overflow:'hidden', borderRadius:40, backgroundColor: 'white'}}
                style={{flex:3, textAlign: 'center', backgroundColor: '#000000', color: 'white', fontSize:20}}
                onPress={() => {Actions.pop();}}>SUBMIT</Button>
          </View>
        </View>
      </DismissableFullscreenPopup>
    )
  }
}

const styles = StyleSheet.create({
  popupView:{
    width: 300,
    height: 200,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'stretch'
  },
  topView:{
    flex: 2,
    backgroundColor: '#1586BA',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },
  topText:{
    color: 'white',
    fontSize: 30
  },
  bottomView:{
    flex: 3,
    backgroundColor: 'white',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },
  bottomText:{
    color: 'black',
    fontSize: 20
  }
});

export default JoinUsPage;
