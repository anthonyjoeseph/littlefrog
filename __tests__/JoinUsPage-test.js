// __tests__/DismissableFullscreenPopup-test.js
import 'react-native';
import React from 'react';
import JoinUsPage from '../app/scenes/modal/JoinUsPage/JoinUsPage';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <JoinUsPage />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
