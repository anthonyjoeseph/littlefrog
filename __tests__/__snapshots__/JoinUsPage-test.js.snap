exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "rgba(0, 0, 0, 0.7)",
      "flexDirection": "row",
      "height": 1334,
      "justifyContent": "center",
      "left": 0,
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": 750,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "stretch",
            "flexDirection": "column",
            "height": 200,
            "justifyContent": "center",
            "width": 300,
          }
        }
        testID={undefined}>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#1586BA",
              "flex": 2,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }>
            Join for Promotions!
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "flex": 3,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 20,
              }
            }>
            Enter your email
          </Text>
          <TextInput
            editable={true}
            style={
              Object {
                "height": 40,
              }
            } />
          <View
            accessibilityComponentType="button"
            accessibilityLabel={undefined}
            accessibilityTraits="button"
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 40,
                "height": 45,
                "opacity": 1,
                "overflow": "hidden",
                "padding": 5,
              }
            }
            testID={undefined}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#007aff",
                    "fontFamily": "HelveticaNeue-Medium",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  null,
                  Object {
                    "backgroundColor": "#000000",
                    "color": "white",
                    "flex": 3,
                    "fontSize": 20,
                    "textAlign": "center",
                  },
                  null,
                ]
              }>
              SUBMIT
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
